src/module/discovery.cpp

```diff
@@ -12,35 +12,92 @@ OrbbecDiscovery::OrbbecDiscovery(vsdk::Dependencies dependencies, vsdk::Resource
-        return {};
+    // Enable network device enumeration for Ethernet cameras (like Gemini 335Le)
+    try {
+        ob_ctx_->enableNetDeviceEnumeration(true);
+        VIAM_RESOURCE_LOG(info) << "Enabled network device enumeration for Ethernet cameras";
+    } catch (ob::Error& e) {
+        VIAM_RESOURCE_LOG(warn) << "Failed to enable network device enumeration: " << e.what();
+    } catch (const std::exception& e) {
+        VIAM_RESOURCE_LOG(warn) << "Failed to enable network device enumeration: " << e.what();
     }
 
-    VIAM_SDK_LOG(info) << "Discovered " << devCount << " devices";
+    try {
+        // Use SDK's native device discovery (works for both USB and network devices)
+        std::shared_ptr<ob::DeviceList> devList = ob_ctx_->queryDeviceList();
+        int devCount = devList->getCount();
+
+        if (devCount == 0) {
+            VIAM_RESOURCE_LOG(warn) << "No Orbbec devices found during discovery";
+            return {};
+        }
+
+        VIAM_RESOURCE_LOG(info) << "Discovered " << devCount << " Orbbec devices";
+
+        for (size_t i = 0; i < devCount; i++) {
+            try {
+                std::string deviceName = devList->name(i);
+                std::string serialNumber = devList->serialNumber(i);
+                std::string connectionType = devList->connectionType(i);
+                std::string ipAddress = devList->ipAddress(i);
 
-    for (size_t i = 0; i < devCount; i++) {
-        std::shared_ptr<ob::Device> dev = devList->getDevice(i);
-        std::shared_ptr<ob::DeviceInfo> info = dev->getDeviceInfo();
+                std::stringstream deviceInfoString;
+                deviceInfoString << "Device " << (i + 1) << " - Name: " << deviceName << ", Serial: " << serialNumber
+                                 << ", Connection: " << connectionType;
+                if (!ipAddress.empty()) {
+                    deviceInfoString << ", IP: " << ipAddress;
+                }
 
-        std::ostringstream name;
-        name << "orbbec-" << i + 1;
+                VIAM_RESOURCE_LOG(info) << deviceInfoString.str();
 
-        vsdk::ProtoStruct attributes;
-        attributes.emplace("serial_number", info->serialNumber());
+                std::ostringstream name;
+                name << "orbbec-" << i + 1;
 
-        vsdk::ResourceConfig config(
-            "camera", std::move(name.str()), "viam", attributes, "rdk:component:camera", orbbec::Orbbec::model, vsdk::log_level::info);
-        configs.push_back(config);
+                vsdk::ProtoStruct attributes;
+                attributes.emplace("serial_number", serialNumber);
+
+                // Detect model and create appropriate resource
+                std::string viamModelSuffix;
+                try {
+                    const auto& modelConfig = orbbec::OrbbecModelConfig::forDevice(deviceName);
+                    viamModelSuffix = modelConfig.viam_model_suffix;
+                } catch (const std::runtime_error& e) {
```

Comment:
we're sure all runtime errors = camera unsupported? any more specific error to catch?